core/lib/brightcontent/base_controller_ext/fields.rb

```diff
@@ -24,6 +25,20 @@ def filter_fields
         []
       end
 
+      # Returns filter field definitions as a nested array.
+      #
+      #  [:foo, { bar: { as: :select }, qux: { as: :string } }]
+      #
+      # Becomes:
+      #
+      #  [[:foo], [:bar, { as: :select }], [:qux, { as: :string }]]
+      #
+      def filter_fields_with_options
+        filter_fields.map do |field, result|
+          field.is_a?(Hash) ? field.to_a : [[field]]
+        end.flatten(1)
+      end
```

Comment:
Also, perhaps a better name for this is `normalize_filter_fields`? Because that's what it does, right?